packages/maskbook/src/utils/native-rpc/types.ts

```diff
@@ -8,7 +8,7 @@ export interface WebviewAPIs {
     getDashboardURL(): Promise<string>
     /** Definition of LaunchPage see https://github.com/DimensionDev/Maskbook/blob/master/packages/maskbook/src/settings/types.ts */
     getSettings(key: 'launchPageSettings'): Promise<LaunchPage>
-    getConnectedPersonas(): Promise<{ network: string; connected: boolean }[][]>
+    getConnectedPersonas(): Promise<{ personas: { network: string; connected: boolean }[][] }>
```

Comment:
Android can't serialize array.

Reply:
interesting. is there any upstream bug link?

Reply:
It's not `cannot serialize array`, it's GeckoView fails to json-fy the output of `getConnectedPersonas`
@zhouhanseng I tried using `Json.stringfy()` to encode the output and it works on Android.
Guess we do not need to add this key, just stringfy the result of this method?

Reply:
why failed to JSON-ify? Is it containing circular reference or it's a bug of GeckoView?

Reply:
Or it's an exotic object (For example, a Proxy, DOM object, Date object, ... that created by the JS engine) / proptotyped object (a class) that cannot be serialized?

Reply:
GeckoView simply logs the error message "fail to dispatch message" on the JS side and hides the underneath error root cause.

Tested on GeckoView debug console as following:
1. `let aport = browser.runtime.connectNative("browser"); ` OK
2. `aport.postMessage([])` OK
3. `aport.postMessage([[]])` `**Error dispatching GeckoView:WebExtension:PortMessage Messaging.jsm:76:24
Invalid event data property data**`

Seems GeckoView cannot JSON-ify an object with nested array

Reply:
😂 @jk234ert please report it to Firefox and leave a link here

Reply:
Bug reported to Mozilla, [link](https://bugzilla.mozilla.org/show_bug.cgi?id=1681638)